augur/tasks/git/facade_tasks.py

```diff
@@ -331,12 +333,50 @@ def git_repo_cleanup_facade_task(repo_git):
         git_repo_cleanup(session, repo_git)
 
 @celery.task
-def git_repo_initialize_facade_task(repo_git):
+def clone_repos():
+
+    logger = logging.getLogger(clone_repos.__name__)
+    
+    is_pending = CollectionStatus.facade_status == CollectionState.PENDING.value
 
-    logger = logging.getLogger(git_repo_initialize_facade_task.__name__)
+    # get first repo to process
 
     with FacadeSession(logger) as session:
-        git_repo_initialize(session, repo_git)
+        try:
+            repo_git = get_collection_status_repo_git_from_filter(session, is_pending, 1)[0]
+        except IndexError:
+            return
+
+        while repo_git:
+        
+            # set repo to intializing
+            repo = session.query(Repo).filter(Repo.repo_git == repo_git).one()
+            repoStatus = repo.collection_status[0]
+            setattr(repoStatus,"facade_status", CollectionState.INITIALIZING.value)
+            session.commit()
+
+            # clone repo
+            git_repo_initialize(session, repo_git)
+
+            # get the commit count
+            commit_count = get_repo_commit_count(session, repo_git)
+            facade_weight = get_facade_weight_with_commit_count(session, repo_git, commit_count)
+
+            update_facade_scheduling_fields(session, repo_git, facade_weight, commit_count)
+
+            # set repo to update
+            repoStatus = repo.collection_status[0]
+            setattr(repoStatus,"facade_status", CollectionState.UPDATE.value)
+            session.commit()
+
+            # get next repo
+            try:
+                repo_git = get_collection_status_repo_git_from_filter(session, is_pending, 1)[0]
+            except IndexError:
+                return
+
```

Comment:
I think this task should re-schedule itself for five minutes in the future and exit in the event that there aren't enough repositories to clone. This way we will always check to clone repos at least every five minutes and we avoid having multiple instances of this task running at once

Reply:
Multiple instances of the same task at once isn't an issue, and it is actually desirable behavior, because if we have enough repos to initialize that a clone task is taking 5 minutes then it would be good to have another task to help finish them faster

Reply:
I just worry about the undefined upper bounds of how many tasks we are scheduling. I think it does make sense to have more than one task working to clone repos at a time but I would prefer if the amount of tasks cloning were predefined or at least had an upper limit of some sort.

Reply:
Okay yeah that makes sense. I wasn't thinking about the case where the collection tasks are using all of the queue and the clone task has to wait until they are done to be processed